src/file_system/mod.rs

```diff
@@ -354,6 +354,23 @@ impl Path {
     }
 }
 
+impl From<Vec<Label>> for Path {
+    fn from(labels: Vec<Label>) -> Self {
+        if labels.len() > 0 {
+            Self::from_labels(labels[0].clone(), &labels[1..])
+        } else {
+            panic!("Cannot convert empty vector to Path")
```

Comment:
In my understanding best practice is to convert all fallible `From` implementations to `TryFrom`. There is even a lint in good ole clippy: https://rust-lang.github.io/rust-clippy/master/#fallible_impl_from

Reply:
Ya, so I would always discourage code like this. `panic` should only be used for code that can truly exit, like a CLI or testing. 

@xla: Is a `TryFrom` instance something application would be interested in?

@massimiliano-mantione: The original issue for the repo was actually to go from `Path` to `Vec<Label>`. I probably worded it badly, so my bad. You'll notice that it's safe to have a function for `Path -> Vec<Label>` without any errors :)

Reply:
> @xla: Is a `TryFrom` instance something application would be interested in?

Maybe, we almost always more likely to come around with a `String` so a `FromStr` sounds like a quality of life improvement.

Reply:
> Maybe, we almost always more likely to come around with a String so a FromStr sounds like a quality of life improvement.

Ya that makes sense, and that was added in #44